src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Int128Converter.cs

```diff
@@ -0,0 +1,48 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Globalization;
+
+namespace System.ComponentModel
+{
+    /// <summary>
+    /// Provides a type converter to convert 128-bit signed integer objects to and
+    /// from various other representations.
+    /// </summary>
+    public class Int128Converter : BaseNumberConverter
+    {
+        /// <summary>
+        /// The Type this converter is targeting (e.g. Int16, UInt32, etc.)
+        /// </summary>
+        internal override Type TargetType => typeof(Int128);
+
+        /// <summary>
+        /// Convert the given value string to Int128 using the given radix
+        /// </summary>
+        internal override object FromString(string value, int radix)
+        {
+            Debug.Assert(radix == 10 && radix == 16);
+            Debug.Assert(value is not null);
+
+            if (radix == 16)
```

Comment:
Since this method is only ever called with `radix == 16`, do we really need this branch? Can we just assert that it is always `16`?

Reply:
Notably the above assert is also wrong, it should be `||` not `&&` at a minimum. But I agree if the `radix == 10` path is dead code, we don't need the comparison.